### DISCUSSION TITLE
Writing a CONTRIBUTING.md

### DISCUSSION THREAD

**Discussion by jacobmealey** [curl/trurl#180] [General]
I've been thinking about writing a contributing file for new contributors. As a newer contributor, there are a few things that I wish I would have known earlier. Some of them may seem trivial to experienced open-source developers but proved a bit tricky for me. It will also be beneficial to have a brief document for some of the internal toolings we use. 

Below are some pre-PR tips I've picked up since starting to work on this project, if folks have other ideas it will be most beneficial.
- run make test, make checksrc, and make test-memory 
- Add tests for new features 
- Document new features in the help prompt and the man page

Also, I'm not sure if this should be added in the README or as a separate CONTRIBUTING.md, but it doesn't seem like many other curl projects have one so I'm not sure what should be done.

**Comment by dfandrich** [comment]
curl is just about the only repository under the curl project that gets
contributions from the outside, and it has docs/CONTRIBUTE.md

**Reply by jacobmealey** [reply]
That makes sense, I was surprised not to see one in curl but I was only looking in the root of the repo I didn't think to check docs/ which now seems obvious to me.

**Comment by bagder** [comment]
I think it sounds like an excellent idea! Keeping these things in a separate file seems sensible to avoid the README from blowing up in size too much.

**Reply by jacobmealey** [reply]
Excellent, I will start writing one today and open a draft PR so folks can provide feedback. I will try to make sure it matches the tone/flow of the contributing file in curl.
